nest/src/interfaces/IVault.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+interface IVault {
```

Comment:
`interface IVault is IERC20`?  Then we wouldn't have to define `transferFrom`, `approve`, `totalSupply`, `balanceOf`, `decimals`. Plus `contract BoringVault is ERC20, Auth`, etc. anyway